File: services/front-end/local-game-pong/src/controller/statistics.js
export default function Statistics() {
	this.score = 0;
	this.touchedBalls = 0;
	this.usedSpells = 0;
	this.timeWithoutTakingGoals = [0, 0];
	this.goalsInARow = [0, 0];
}

Statistics.prototype.increaseScore = function () {
	this.score++;
}

Statistics.prototype.increaseTouchedBall = function () {
	this.touchedBalls++;
}

Statistics.prototype.increaseUsedSpells = function () {
	this.usedSpells++;
}

Statistics.prototype.increaseTimeWithoutTakingGoals = function () {
	this.timeWithoutTakingGoals[1]++;
}

Statistics.prototype.resetTimeWithoutTakingGoals = function () {
	if (this.timeWithoutTakingGoals[1] > this.timeWithoutTakingGoals[0])
		this.timeWithoutTakingGoals[0] = this.timeWithoutTakingGoals[1];
	this.timeWithoutTakingGoals[1] = 0;
}

Statistics.prototype.resetGoalsInARow = function () {
	if (this.goalsInARow[1] > this.goalsInARow[0])
		this.goalsInARow[0] = this.goalsInARow[1];
	this.goalsInARow[1] = 0;
}

Statistics.prototype.increaseGoalsInARow = function () {
	this.goalsInARow[1]++;
}

Statistics.prototype.getScore = function () {
	return this.score;
}

Statistics.prototype.getAllStatisticsAsArray = function () {
	return [this.score, this.touchedBalls, this.usedSpells, this.timeWithoutTakingGoals[0], this.goalsInARow[0]];
}

File: services/back-game-pong/server/statistics.py
class Statistics:
	def __init__(self):
		self.score = 0
		self.touched_balls = 0
		self.used_spells = 0
		self.time_without_taking_goals = [0, 0]
		self.goals_in_row = [0, 0]

	def increase_time_without_taking_goals(self):
		self.time_without_taking_goals[1] += 1

	def increase_goals_in_row(self):
		self.goals_in_row[1] += 1

	def reset_time_without_taking_goals(self):
		if self.time_without_taking_goals[1] > self.time_without_taking_goals[0]:
			self.time_without_taking_goals[0] = self.time_without_taking_goals[1]
			self.time_without_taking_goals[1] = 0

	def reset_goals_in_row(self):
		if self.goals_in_row[1] > self.goals_in_row[0]:
			self.goals_in_row[0] = self.goals_in_row[1]
			self.goals_in_row[1] = 0

	def get_statistics_as_list(self):
		return [self.score, self.touched_balls, self.used_spells, self.time_without_taking_goals[0], self.goals_in_row[0]]
